### PR TITLE
Deprecate node_metadata and add mode

### PR DESCRIPTION
**Pull request by c2thorn**
Part of https://github.com/hashicorp/terraform-provider-google/issues/8593

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:deprecation
container: deprecated `workload_metadata_configuration.node_metadata` in favor of `workload_metadata_configuration.mode` in `google_container_cluster`
```

```release-note:enhancement
container: added support for `workload_metadata_configuration.mode` in `google_container_cluster`
```

### CODE REVIEW

**Comment by c2thorn** [comment]
Tests passed in the previous run, I just added the upgrade guide note

File: mmv1/third_party/terraform/tests/resource_container_node_pool_test.go.erb [hunk 1]
```diff
@@ -205,6 +205,50 @@ func TestAccContainerNodePool_withWorkloadIdentityConfig(t *testing.T) {
...
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("google_container_node_pool.with_workload_metadata_config",
+						"node_config.0.workload_metadata_config.0.mode", "GCE_METADATA"),
+				),
+			},
+			{
+				ResourceName:      "google_container_node_pool.with_workload_metadata_config",
+				ImportState:       true,
+				ImportStateVerify: true,
+				// Import always uses the v1 API, so beta features don't get imported.
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/resource_container_node_pool_test.go.erb:217]: Wait, what? GKE uses the beta client, at least as of this change.

File: mmv1/third_party/terraform/website/docs/r/container_cluster.html.markdown [hunk 2]
```diff
@@ -868,15 +868,21 @@ The `taint` block supports:
...
 
-The `workload_metadata_config` block supports:
+The `workload_metadata_config` must have exactly one of `node_metadata` (deprecated) or `mode` set. This block supports:
 
-* `node_metadata` (Required) How to expose the node metadata to the workload running on the node.
+* `node_metadata` (Optional, Deprecated) How to expose the node metadata to the workload running on the node. This is deprecated in favor of `mode`
     Accepted values are:
     * UNSPECIFIED: Not Set
-    * SECURE: Prevent workloads not in hostNetwork from accessing certain VM metadata, specifically kube-env, which contains Kubelet credentials, and the instance identity token. See [Metadata Concealment](https://cloud.google.com/kubernetes-engine/docs/how-to/metadata-proxy) documentation.
+    * SECURE: Prevent workloads not in hostNetwork from accessing certain VM metadata, specifically kube-env, which contains Kubelet credentials, and the instance identity token. See [Metada Concealment](https://cloud.google.com/kubernetes-engine/docs/how-to/metadata-proxy) documentation.
```
Comment (rileykarson) [inline mmv1/third_party/terraform/website/docs/r/container_cluster.html.markdown:876]: `Metadata` - > `Metada` seems unintentional?

File: mmv1/third_party/terraform/utils/node_config.go.erb [hunk 3]
```diff
@@ -265,9 +267,17 @@ func schemaNodeConfig() *schema.Schema {
...
+								Optional:     true,
+								Computed:     true,
+								Deprecated:   "Deprecated in favor of mode.",
 								ValidateFunc: validation.StringInSlice([]string{"UNSPECIFIED", "SECURE", "EXPOSE", "GKE_METADATA_SERVER"}, false),
-								Description: `NodeMetadata is the configuration for how to expose metadata to the workloads running on the node.`,
+								Description:  `NodeMetadata is the configuration for how to expose metadata to the workloads running on the node.`,
+							},
+							"mode": {
+								Type:         schema.TypeString,
+								Optional:     true,
```
Comment (rileykarson) [inline mmv1/third_party/terraform/utils/node_config.go.erb:278]: Will the API return this field if `node_metadata` is set? We probably need `Computed: true` if so.

Comment (c2thorn) [reply mmv1/third_party/terraform/utils/node_config.go.erb:278]: I just tested applying `mode` first, and then replacing with `node_metadata`. It did not return `mode`.

File: mmv1/third_party/terraform/tests/resource_container_node_pool_test.go.erb (hunk 1 above)
Comment (c2thorn) [reply mmv1/third_party/terraform/tests/resource_container_node_pool_test.go.erb]: This was a copy-paste from a similar test, but it seems like this comment and the verify ignore no longer applies. I've removed them from both.

**Review by rileykarson — APPROVED** [review]

**Pull request merged**